modules/apps/27-interchain-accounts/host/keeper/msg_server.go

```diff
@@ -0,0 +1,66 @@
+package keeper
+
+import (
+	"context"
+
+	errorsmod "cosmossdk.io/errors"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	abci "github.com/cometbft/cometbft/abci/types"
+
+	"github.com/cosmos/ibc-go/v7/modules/apps/27-interchain-accounts/host/types"
+	ibcerrors "github.com/cosmos/ibc-go/v7/modules/core/errors"
+)
+
+var _ types.MsgServer = (*msgServer)(nil)
+
+type msgServer struct {
+	*Keeper
+}
+
+// NewMsgServerImpl returns an implementation of the ICS27 host MsgServer interface
+// for the provided Keeper.
+func NewMsgServerImpl(keeper *Keeper) types.MsgServer {
+	return &msgServer{Keeper: keeper}
+}
```

Comment:
do we wanna do the same thing here with panicing if `queryRouter` is nil to enforce the feature is picked up?

Reply:
Yes, we want. I will add it.